### ISSUE TITLE
Conflict with Not Enough Crashes

### ISSUE THREAD

**Issue by Fourmisain** [chirs241097/jline4mcdsrv#5]
Refering to https://github.com/natanfudge/Not-Enough-Crashes/issues/28, jline4mcdsrv and NEC previously had a Mixin conflict that is now resolved.

However, if both mods are run together, all log lines will be doubled once jline4mcdsrv is done with it's logger changes.

This most definitely has to do with this piece of code:
https://github.com/chirs241097/jline4mcdsrv/blob/a13f90543e816d5c3eb623b2efe33bc8bf9b5cdf/src/main/java/org/chrisoft/jline4mcdsrv/Console.java#L51-L56

More specifically, I believe it's still (like in https://github.com/natanfudge/Not-Enough-Crashes/issues/28) this line which fails:
https://github.com/chirs241097/jline4mcdsrv/blob/a13f90543e816d5c3eb623b2efe33bc8bf9b5cdf/src/main/java/org/chrisoft/jline4mcdsrv/Console.java#L54

Somehow the default "SysOut" logger can't be found, thus is not removed and so we get double the log lines.

**Comment by Fourmisain** [comment]
I tried adding support for NEC in my https://github.com/Fourmisain/jline4mcdsrv/tree/nec_compat branch, which I based on #4 - I'll create another pull request once that's merged.

The issue was that NEC [installs a RewritePolicy](https://github.com/natanfudge/Not-Enough-Crashes/blob/d609012e61d7c37f1a807c6cddfe65c884b1e4b4/notenoughcrashes/src/main/java/fudge/notenoughcrashes/DeobfuscatingRewritePolicy.java#L17-L41) by removing all base appenders and wrapping them in a new `RewriteAppender` with the name `NotEnoughCrashesDeobfuscatingAppender`.

jline4mcdsrv tries to replace `SysOut` by removing the appender and adding it's own, but `SysOut` is already (re)moved, so we end up with 2 appenders, both logging to the console.

The fix I implemented was to detect the `NotEnoughCrashesDeobfuscatingAppender`, reading it's `RewritePolicy` (which there was no other way than using Reflections, because Log4J doesn't think a getter is a sensible option...), using it inside jline4mcdsrv's "Console" appender and redoing the whole NEC "installation" step, this time without `SysOut` (again because I didn't find an easier way with Log4J).

Is it hacky? Yes.
Does it work? Maybe.
I didn't actually test the stack trace deobfuscation, I only think it _should_ work.
